android/app/src/main/java/org/fossasia/openevent/adapters/viewholders/DayScheduleViewHolder.java

```diff
@@ -89,50 +94,34 @@ public void bindSession(RealmDataRepository realmRepo){
             slotTrack.getBackground().setColorFilter(storedColor, PorterDuff.Mode.SRC_ATOP);
             slotTrack.setText(sessionTrack.getName());
 
-            if(session.getIsBookmarked()) {
+            if (session.getIsBookmarked()) {
                 slot_bookmark.setImageResource(R.drawable.ic_bookmark_white_24dp);
             } else {
                 slot_bookmark.setImageResource(R.drawable.ic_bookmark_border_white_24dp);
             }
-            slot_bookmark.setColorFilter(storedColor,PorterDuff.Mode.SRC_ATOP);
+            slot_bookmark.setColorFilter(storedColor, PorterDuff.Mode.SRC_ATOP);
 
             final int sessionId = session.getId();
 
             slot_bookmark.setOnClickListener(v -> {
-                if(session.getIsBookmarked()) {
+                if (session.getIsBookmarked()) {
 
                     realmRepo.setBookmark(sessionId, false).subscribe();
                     slot_bookmark.setImageResource(R.drawable.ic_bookmark_border_white_24dp);
-
-                    if ("MainActivity".equals(context.getClass().getSimpleName())) {
-                        Snackbar snackbar = Snackbar.make(slot_content, R.string.removed_bookmark, Snackbar.LENGTH_LONG)
-                                .setAction(R.string.undo, view -> {
-
-                                    realmRepo.setBookmark(sessionId, true).subscribe();
-                                    slot_bookmark.setImageResource(R.drawable.ic_bookmark_white_24dp);
-
-                                    WidgetUpdater.updateWidget(context);
-                                })
-                                .setActionTextColor(ColorUtils.setAlphaComponent(Views.getAccentColor(context), 220));
-                        snackbar.show();
-                    } else {
-                        Snackbar.make(slot_content, R.string.removed_bookmark, Snackbar.LENGTH_SHORT).show();
-                    }
+                    dayScheduleCallback.showSnackbar(R.string.removed_bookmark, Snackbar.LENGTH_LONG, CODE_UNDO_REMOVED, sessionId, slot_bookmark, storedColor);
```

Comment:
Check for null before calling the callback

Reply:
Please reduce the arguments amount to at most 2. We don't need to pass the snack bar length and string resource to be used. It should just have id and information that it was removed or added, which can be expressed through a boolean

Reply:
@iamareebjamal We will still need the reference to slot_bookmark right? Should I pass the view holder itself as then I dont need to pass color, session etc

Reply:
Create an Object with these parameters and send it instead. Sounds good?

Reply:
Yep cool